dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/bundle/TrackedEntityDataValueAuditTest.java

```diff
@@ -120,6 +120,7 @@ void testTrackedEntityDataValueAuditCreate()
             assertEquals( a.getProgramStageInstance().getUid(), psi.getUid() );
         } );
 
+        assertEquals( 1, updatedAudit.size() );
```

Comment:
We are regularly having issues with tests failing on master only on CI with little to no context.

We should absolutely avoid assertions like this for this reason. Imagine this test fails on GitHub only telling you `expected 1 got 2`. What are you going to do? 😅 Multiple devs lost a few days on multiple occasions trying to debug tests with such assertions 😬 

Please use for example
https://junit.org/junit5/docs/5.8.2/api/org.junit.jupiter.api/org/junit/jupiter/api/Assertions.html#assertAll(java.lang.String,java.util.Collection)

like in https://github.com/dhis2/dhis2-core/blob/0d367e9b804ae5476694ab9545840e21b4cd68e5/dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/trigramsummary/TrigramSummaryControllerTest.java#L243-L245

to group the assertion for checking the size and the content below into one assertAll

This would allow us to at least know that if the `updatedAudit` contains 3 elements if the one we expect is in there.

Also try failing the test locally. Would you know what do do with the error message if it were to fail on GitHub only?

Adding a supplier to JUnit assertions like
https://github.com/dhis2/dhis2-core/blob/6e83eba877030575456f919de1bce55fcae8e517/dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/tracker/Assertions.java#L44-L59

helps a lot with adding context, in case the assertion itself does not do that for us.

Reply:
resolved.